lib/pysquared/Config.py

```diff
@@ -0,0 +1,80 @@
+"""
+Class for encapsulating config.json. The goal is to
+distribute these values across the files & variables
+that use them. Instantiation happens in main.
+
+Also it allow values to be set temporarily using setter
+function, and values can be set permanently using the
+saver functions. Following the FPrime model.
+"""
+
+import json
+
+import lib.pysquared.commandsConfig as commandsConfig
+
+
+class Config:
+    """
+    Constructor
+    """
+
+    def __init__(self) -> None:
+        # parses json & assigns data to variables
+        with open("config.json", "r") as f:
+            json_data = f.read()
+        self._config: dict = json.loads(json_data)
+
+    """
+    Categorized getter functions
+    """
+
+    def getStrValue(self, strVariable: str) -> str:
+        return self._config[strVariable]
+
+    def getIntValue(self, intVariable: str) -> int:
+        return self._config[intVariable]
+
+    def getFloatValue(self, floatVariable: str) -> float:
+        return self._config[floatVariable]
+
+    def getBoolValue(self, boolVariable: str) -> bool:
+        return self._config[boolVariable]
+
+    def getListValue(self, listVariable: str) -> list[str]:
+        return self._config[listVariable]
```

Comment:
I like this for type hinting purposes but since there is no type enforcement at runtime I'm a little dubious of the value add over a single `def get(self, key: str) -> Union(str, int, float, bool, list...)`. The single function would make this module easier to maintain in the long run. Not a blocker but something to think on and keep an eye on after you get this version merged.

In the near term, what do you think of this naming modification?
```suggestion
    def getStr(self, key: str) -> str:
        """Gets a string value from the config dictionary"""
        return self._config[key]

    def getInt(self, key: str) -> int:
        """Gets an int value from the config dictionary"""
         return self._config[key]

    def getFloat(self, key: str) -> float:
        """Gets a float value from the config dictionary"""
        return self._config[key]

    def getBool(self, key: str) -> bool:
        """Gets a bool value from the config dictionary"""
        return self._config[key]

    def getList(self, key: str) -> list[str]:
        """Gets a list value from the config dictionary"""
        return self._config[key]
```

Reply:
Yup this sounds good to me. One of the things I was aiming for here was keeping the code as clean and simple as possible. Your suggested function does a good job at this.